common/crypto/expiration.go

```diff
@@ -34,3 +38,60 @@ func ExpiresAt(identityBytes []byte) time.Time {
 	}
 	return cert.NotAfter
 }
+
+// WarnFunc notifies a warning happened with the given format, and can be replaced with Warnf of a logger.
+type WarnFunc func(format string, args ...interface{})
+
+// Scheduler invokes f after d time, and can be replaced with time.AfterFunc.
+type Scheduler func(d time.Duration, f func()) *time.Timer
+
+// TrackExpiration warns a week before one of the certificates expires
+func TrackExpiration(tls bool, serverCert []byte, clientCertChain [][]byte, sIDBytes []byte, warn WarnFunc, now time.Time, s Scheduler) {
```

Comment:
This is an interesting section to see come into the code base...

Reply:
nice